shoop/front/middleware.py

```diff
@@ -91,15 +87,22 @@ def _set_timezone(self, request):
             timezone.activate(request.person.timezone)
             # TODO: Fallback to request.shop.timezone (and add such field)
 
-    def _get_maintenance_response(self, request):
-        if request.shop.maintenance_mode and not request.user.is_superuser:
-            return HttpResponse(loader.render_to_string("shoop/front/maintenance.jinja", request=request), status=503)
-
     def process_response(self, request, response):
         if hasattr(request, "basket") and request.basket.dirty:
             request.basket.save()
+        maintenance_response = self._get_maintenance_response(request)
```

Comment:
Could we use [process_view](https://docs.djangoproject.com/en/1.8/topics/http/middleware/#process-view) instead?  Then we'd have `resolver_match` available, and we could even introspect the `view` argument (I assume the point here is to allow logging in to maintenance mode shops) to see if it "smells like" a login view.